src/main/java/org/opentripplanner/routing/algorithm/RoutingWorker.java

```diff
@@ -128,6 +130,16 @@ public RoutingResponse route() {
     boolean removeWalkAllTheWayResultsFromDirectFlex =
       request.journey().direct().mode() == StreetMode.FLEXIBLE;
 
+    var modes = request.journey().modes();
+    boolean hasBikePark = List
+      .of(modes.accessMode, modes.egressMode)
+      .contains(StreetMode.BIKE_TO_PARK);
+
+    double minBikeParkingDistance = NOT_SET;
+    if (hasBikePark) {
+      minBikeParkingDistance = request.preferences().itineraryFilter().minBikeParkingDistance();
+    }
```

Comment:
Can we extract this, not to make the already complex method any mor complex. I'm also not really happy about using `org.opentripplanner.routing.algorithm.filterchain.ItineraryListFilterChainBuilder.NOT_SET` from here, can we find a better solution?

Reply:
That sounds like a good idea